unsafe/src/main/java/org/apache/spark/unsafe/PlatformDependent.java

```diff
@@ -145,21 +147,27 @@ public static void freeMemory(long address) {
 
       long signumOffset = 0;
       long magOffset = 0;
+      long intCompactOffset = 0;
       try {
         signumOffset = _UNSAFE.objectFieldOffset(BigInteger.class.getDeclaredField("signum"));
         magOffset = _UNSAFE.objectFieldOffset(BigInteger.class.getDeclaredField("mag"));
+        intCompactOffset = _UNSAFE.objectFieldOffset(
+          BigDecimal.class.getDeclaredField("intCompact"));
```

Comment:
It's a little scary to be relying on implementation details of `BigDecimal` here, but I suppose that this would fail-fast with an exception if `BigDecimal` lacked that field.